### ISSUE TITLE
Loss does not decrease

### ISSUE THREAD

**Issue by pandasMX** [ppwwyyxx/tensorpack#634]
Hi,

I can run your resnet-keras example. Then i modified the code and changed the net to mobilenet. It can run but the loss does not decrease and accuracy close to zero. I compared the log between resnet-keras and my mobilenet implementation. Turned out that for the mobilenet, there's no total loss showing in the log. Do you have any suggestions? Thanks!

mobilenet:
[0207 01:46:45 @monitor.py:363] DataParallelInferenceRunner/QueueInput/queue_size: 2.5896
[0207 01:46:45 @monitor.py:363] GPUUtil/0: 91.15
[0207 01:46:45 @monitor.py:363] GPUUtil/1: 91.383
[0207 01:46:45 @monitor.py:363] GPUUtil/2: 91.567
[0207 01:46:45 @monitor.py:363] GPUUtil/3: 91.623
[0207 01:46:45 @monitor.py:363] QueueInput/queue_size: 49.033
[0207 01:46:45 @monitor.py:363] categorical_accuracy: 0.0065316
[0207 01:46:45 @monitor.py:363] categorical_crossentropy: 4.7449
[0207 01:46:45 @monitor.py:363] lr: 0.1
[0207 01:46:45 @monitor.py:363] validation_categorical_accuracy: 0.0076488
[0207 01:46:45 @group.py:42] Callbacks took 23.882 sec in total. DataParallelInferenceRunner: 23.821sec
[0207 01:46:45 @base.py:247] Start Epoch 4 ...

resnet:
[0131 03:42:25 @monitor.py:363] DataParallelInferenceRunner/QueueInput/queue_size: 48.586
[0131 03:42:25 @monitor.py:363] GPUUtil/0: 88.415
[0131 03:42:25 @monitor.py:363] GPUUtil/1: 89.214
[0131 03:42:25 @monitor.py:363] GPUUtil/2: 89.876
[0131 03:42:25 @monitor.py:363] GPUUtil/3: 89.787
[0131 03:42:25 @monitor.py:363] QueueInput/queue_size: 49.43
[0131 03:42:25 @monitor.py:363] categorical_accuracy: 0.26678
[0131 03:42:25 @monitor.py:363] categorical_crossentropy: 2.9259
[0131 03:42:25 @monitor.py:363] lr: 0.1
[0131 03:42:25 @monitor.py:363] regularize_cost: 0.40948
[0131 03:42:25 @monitor.py:363] total_loss: 3.3354
[0131 03:42:25 @monitor.py:363] validation_categorical_accuracy: 0.30536
[0131 03:42:25 @group.py:42] Callbacks took 29.288 sec in total. DataParallelInferenceRunner: 29.054sec
[0131 03:42:25 @base.py:247] Start Epoch 4 ...

**Comment by pandasMX** [comment]
the mobilenet code is from official keras implementation, the only change i made is the following function:
`def mobilenet(inputs):
    input = tf.layers.Input(tensor=inputs[0])

    def image_preprocess(image):
        image = ImageNetModel.image_preprocess(image)
        image = tf.transpose(image, [0, 3, 1, 2])
        return image

    x = Lambda(image_preprocess)(input)
    
    x = _conv_block(x, 32, ALPHA, strides=(2, 2))
    x = _depthwise_conv_block(x, 64, ALPHA, DEPTH_MULTIPLIER, block_id=1)

    x = _depthwise_conv_block(x, 128, ALPHA, DEPTH_MULTIPLIER,
                              strides=(2, 2), block_id=2)
    x = _depthwise_conv_block(x, 128, ALPHA, DEPTH_MULTIPLIER, block_id=3)

    x = _depthwise_conv_block(x, 256, ALPHA, DEPTH_MULTIPLIER,
                              strides=(2, 2), block_id=4)
    x = _depthwise_conv_block(x, 256, ALPHA, DEPTH_MULTIPLIER, block_id=5)

    x = _depthwise_conv_block(x, 512, ALPHA, DEPTH_MULTIPLIER,
                              strides=(2, 2), block_id=6)
    x = _depthwise_conv_block(x, 512, ALPHA, DEPTH_MULTIPLIER, block_id=7)
    x = _depthwise_conv_block(x, 512, ALPHA, DEPTH_MULTIPLIER, block_id=8)
    x = _depthwise_conv_block(x, 512, ALPHA, DEPTH_MULTIPLIER, block_id=9)
    x = _depthwise_conv_block(x, 512, ALPHA, DEPTH_MULTIPLIER, block_id=10)
    x = _depthwise_conv_block(x, 512, ALPHA, DEPTH_MULTIPLIER, block_id=11)

    x = _depthwise_conv_block(x, 1024, ALPHA, DEPTH_MULTIPLIER,
                              strides=(2, 2), block_id=12)
    x = _depthwise_conv_block(x, 1024, ALPHA, DEPTH_MULTIPLIER, block_id=13)

    shape = (int(1024 * ALPHA), 1, 1)

    x = GlobalAveragePooling2D()(x)
    x = Reshape(shape, name='reshape_1')(x)
    x = Dropout(0.001, name='dropout')(x)
    x = Conv2D(NUM_CLASSES, (1, 1),
               padding='same', name='conv_preds')(x)
    x = Activation('softmax', name='act_softmax')(x)
    x = Reshape((NUM_CLASSES,), name='reshape_2')(x)  
    
    M = tf.keras.models.Model(input, x, name='mobilenet_0.75_160')
    return M`

**Comment by ppwwyyxx** [comment]
Total loss didn't show up because I forgot to add its summary when no regularizer is given. Will add it.

Loss not decreasing is not a tensorpack issue unless you believe there is a tensorpack bug causing this.
